python/pyspark/sql/types.py

```diff
@@ -1137,7 +1148,28 @@ def _verify_type(obj, dataType):
         if type(obj) not in _acceptable_types[_type]:
             raise TypeError("%s can not accept object %r in type %s" % (dataType, obj, type(obj)))
 
-    if isinstance(dataType, ArrayType):
+    if isinstance(dataType, ByteType):
+        if obj < -128 or obj > 127:
+            raise ValueError("object of ByteType out of range, got: %s" % obj)
+
+    elif isinstance(dataType, ShortType):
+        if obj < -32768 or obj > 32767:
+            raise ValueError("object of ShortType out of range, got: %s" % obj)
+
+    elif isinstance(dataType, IntegerType):
+        if obj < -2147483648 or obj > 2147483647:
+            raise ValueError("object of IntegerType out of range, got: %s" % obj)
+
+    elif isinstance(dataType, FloatType):
+        from math import isinf
+        from struct import pack, unpack
+
+        if not isinf(obj):
+            f = unpack("f", pack("f", obj))[0]
```

Comment:
I'm worried that this will be super slow, I think we don't need this.

If overflow happens, it will become infinity.